nvaccess/nvda
# Fix line number reporting in Visual Studio

### Link to issue number:
Closes #13574 

### Summary of the issue:
In Visual Studio, when line number reporting is enabled in VS itself, line numbers are part of the text. Therefore they are reported regardless of NVDA's line reporting settings. This also causes indentation reporting to fail.

### Description of how this pull request fixes the issue:
This pr isolates the line number from the text and ensures that it is reported appropriately when line number reporting in NVDA is on.

### Testing strategy:
With line reporting enabled in Visual studio
* Ensure that line numbers are reported when enabled in NVDA
* Ensure that line numbers are not reported when disabled in NVDA

### Known issues with pull request:
Line number reporting does not work when line numbers are disabled in Visual Studio.

### Change log entries:
Bug fixes
- When Visual Studio is set up to show line numbers in the editor, NVDA's setting to report line numbers is now respected.
 - This also fixes line indentation not being reported correctly. (#13574)
     - Note that for line number reporting to work, showing line numbers must still be enabled in Visual Studio.
 -

### Code Review Checklist:



  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
